File: library/src/main/java/com/imangazaliev/circlemenu/CircleMenuButton.java
package com.imangazaliev.circlemenu;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.StateListDrawable;
import android.graphics.drawable.shapes.OvalShape;
import android.os.Build;
import android.os.Build.VERSION_CODES;
import android.support.v7.widget.AppCompatImageButton;
import android.util.AttributeSet;
import android.util.Log;

public class CircleMenuButton extends CircleButton {

    private int colorNormal;
    private String hintText;

    public CircleMenuButton(Context context, AttributeSet attrs) {
        super(context, attrs);

        TypedArray attr = context.obtainStyledAttributes(attrs, R.styleable.CircleMenuButton, 0, 0);
        colorNormal = attr.getColor(R.styleable.CircleMenuButton_colorNormal, getResources().getColor(R.color.circle_menu_button_color_normal));
        int colorPressed = attr.getColor(R.styleable.CircleMenuButton_colorPressed, getResources().getColor(R.color.circle_menu_button_color_pressed));
        hintText = attr.getString(R.styleable.CircleMenuButton_hintText);
        int iconId = attr.getResourceId(R.styleable.CircleMenuButton_icon, 0);
        attr.recycle();

        setBackgroundCompat(createBackgroundDrawable(colorNormal, colorPressed));

        if (iconId != 0) {
            setImageResource(iconId);
        } else {
            setImageDrawable(new ColorDrawable(Color.TRANSPARENT));
        }
    }

    int getColorNormal() {
        return colorNormal;
    }

    public String getHintText() {
        return hintText;
    }



}

File: library/src/main/java/com/imangazaliev/circlemenu/CircleMenuButton.kt
package com.imangazaliev.circlemenu

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Color
import android.util.AttributeSet
import androidx.annotation.DrawableRes
import androidx.core.content.ContextCompat
import com.google.android.material.floatingactionbutton.FloatingActionButton


class CircleMenuButton @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null
) : FloatingActionButton(context, attrs) {

    internal var iconResId: Int = -1
    private var iconColor: Int = Color.WHITE

    init {
        isClickable = true
    }

    internal fun setIcon(@DrawableRes iconResId: Int) {
        this.iconResId = iconResId

        val icon = ContextCompat.getDrawable(context, iconResId)!!
        icon.setTintCompat(iconColor)
        setImageDrawable(icon)
    }

    fun setIconColor(buttonIconsColor: Int) {
        this.iconColor = buttonIconsColor
    }

    internal fun setColor(color: Int) {
        backgroundTintList = ColorStateList.valueOf(color)
    }

}